opentelemetry-otlp/src/exporter/mod.rs

```diff
@@ -250,7 +251,7 @@ impl<B: HasExportConfig> WithExportConfig for B {
     }
 
     fn with_timeout(mut self, timeout: Duration) -> Self {
-        self.export_config().timeout = timeout;
+        self.export_config().timeout = Some(timeout);
```

Comment:
Please modify the doc comment for all these methods to indicate that they are final and won't be overrrideable via ENV.